subprojects/plugins/src/main/java/org/gradle/api/plugins/JvmTestSuitePlugin.java

```diff
@@ -49,8 +55,8 @@ public void apply(Project project) {
         JavaPluginExtension java = project.getExtensions().getByType(JavaPluginExtension.class);
         TestingExtension testing = project.getExtensions().getByType(TestingExtension.class);
         ExtensiblePolymorphicDomainObjectContainer<TestSuite> testSuites = testing.getSuites();
-        testSuites.registerFactory(TestSuite.class, name -> project.getObjects().newInstance(DefaultJvmTestSuite.class, name, project, java));
-        testSuites.registerFactory(JvmTestSuite.class, name -> project.getObjects().newInstance(DefaultJvmTestSuite.class, name, project, java));
+        testSuites.registerBinding(TestSuite.class, DefaultJvmTestSuite.class);
```

Comment:
I see in the Javadoc that this is preferable, but why?  What does it add?  How should I understand where the "name" and "java" arguments are injected from?